imdbdrinks_ratingsmodule/Repositories/DatabaseReviewRepository.cs

```diff
@@ -1,172 +1,147 @@
+                    UserId = reader.GetInt32(reader.GetOrdinal("UserId")),
+                    Content = reader.GetString(reader.GetOrdinal("Content")),
+                    CreationDate = reader.GetDateTime(reader.GetOrdinal("CreationDate")),
+                    IsActive = reader.GetBoolean(reader.GetOrdinal("IsActive")),
+                };
             }
 
-            return null;
+            throw new Exception($"Review with ID {reviewId} not found.");
         }
 
-        public IEnumerable<Review> FindByRatingId(long ratingId)
+        public IEnumerable<Review> GetReviewsByRatingId(int ratingId)
         {
             var reviews = new List<Review>();
 
-            using (var connection = new SqlConnection(_connectionString))
+            using var connection = this.databaseConnection.CreateConnection();
+            connection.Open();
+
+            using var command = new SqlCommand(ReviewQueries.GetReviewsByRatingIdQuery, connection);
+            command.Parameters.AddWithValue("@RatingId", ratingId);
+
+            using var reader = command.ExecuteReader();
+            while (reader.Read())
             {
-                connection.Open();
-                using (var command = new SqlCommand("SELECT * FROM Reviews WHERE RatingId = @RatingId", connection))
+                reviews.Add(new Review
                 {
-                    command.Parameters.AddWithValue("@RatingId", ratingId);
-
-                    using (var reader = command.ExecuteReader())
-                    {
-                        while (reader.Read())
-                        {
-                            reviews.Add(new Review
-                            {
-                                ReviewId = reader.GetInt32(reader.GetOrdinal("ReviewId")),
-                                RatingId = reader.GetInt32(reader.GetOrdinal("RatingId")),
-                                UserId = reader.GetInt32(reader.GetOrdinal("UserId")),
-                                Content = reader.GetString(reader.GetOrdinal("Content")),
-                                CreationDate = reader.GetDateTime(reader.GetOrdinal("CreationDate")),
-                                IsActive = Convert.ToBoolean(reader["IsActive"])
-                            });
-                        }
-                    }
-                }
+                    ReviewId = reader.GetInt32(reader.GetOrdinal("ReviewId")),
+                    RatingId = reader.GetInt32(reader.GetOrdinal("RatingId")),
+                    UserId = reader.GetInt32(reader.GetOrdinal("UserId")),
+                    Content = reader.GetString(reader.GetOrdinal("Content")),
+                    CreationDate = reader.GetDateTime(reader.GetOrdinal("CreationDate")),
+                    IsActive = Convert.ToBoolean(reader["IsActive"]),
+                });
             }
 
             return reviews;
         }
 
-        public Review Save(Review review)
+        public Review AddOrUpdateReview(Review review)
```

Comment:
Why add or update in same function, doesnt respect single responsibility principle. Separate in two functions